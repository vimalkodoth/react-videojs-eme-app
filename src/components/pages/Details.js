import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchMovieDetail } from "./../../actionCreators";
import DetailsHeader from "./../DetailsHeader";
import Button from "./../../components/Button";
import withBackButton from "./../hocs/withBackButton";
import { func, object } from "prop-types";

class Details extends Component {
    static propTypes = {
        onBackButtonClicked: func,
        fetchMovieDetail: func,
        movieDetails: object,
        match: object
    };
    static defaultParams = {
        onBackButtonClicked: (n) => n,
        fetchMovieDetail: (n) => n,
        match: {},
        movieDetails: {}
    };
    componentDidMount() {
        const {
            match: { params }
        } = this.props;
        this.props.fetchMovieDetail(params.id);
    }

    goBack = (e) => {
        this.props.onBackButtonClicked(e);
    };

    render() {
        const { match } = this.props;
        return (
            <div className="details">
                <Button onClick={this.goBack} show={true} prev={true} />
                <DetailsHeader id={match.params.id} />
            </div>
        );
    }
}

const mapStateToProps = (state) => ({ movieDetails: state.movieDetails });
const mapDispatchToProps = (dispatch) => ({
    fetchMovieDetail(id) {
        dispatch(fetchMovieDetail(id));
    }
});

export const UnWrappedDetails = Details;

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withBackButton(Details));
