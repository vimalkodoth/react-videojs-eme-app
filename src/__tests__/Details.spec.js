import React from "react";
import { render } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Details from "../components/pages/Details";
import detailsItem from "./json/details-item.json";
let store;
const mockStore = configureStore([]);
const originalConsoleError = console.error;
describe("Details", () => {
    beforeEach(() => {
        console.error = jest.fn();
        store = mockStore({
            movieDetails: {
                "el-avion-del-dinero": detailsItem
            }
        });
    });

    afterEach(() => {
        console.error = originalConsoleError;
        store = mockStore({});
    });

    it("renders correctly", () => {
        store.dispatch = jest.fn();
        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Details
                        match={{
                            params: { id: "el-avion-del-dinero" },
                            isExact: true,
                            path: "",
                            url: ""
                        }}
                    />
                </MemoryRouter>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("renders back button", () => {
        store.dispatch = jest.fn();
        const component = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Details
                        match={{
                            params: { id: "el-avion-del-dinero" },
                            isExact: true,
                            path: "",
                            url: ""
                        }}
                    />
                </MemoryRouter>
            </Provider>
        );
        expect(component.find('[data-testid="button"]')).toBeTruthy();
        expect(component.find('a[href="/trailer"]')).toBeTruthy();
        expect(
            component.find('[data-testid="header-meta"] .title').text()
        ).toEqual("El Avi√≥n del Dinero");
    });
});
