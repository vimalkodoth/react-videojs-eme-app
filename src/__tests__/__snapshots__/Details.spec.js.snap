// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <MemoryRouter>
    <Connect(Sub)
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": "el-avion-del-dinero",
          },
          "path": "",
          "url": "",
        }
      }
    />
  </MemoryRouter>
</ContextProvider>
`;
